tests/pending/pos/blueSkyExtensions1.scala

```diff
@@ -0,0 +1,134 @@
+/** A blue sky sketch how one might evolve extensions to do type classes
+ *  without the current gap between functional and OO patterns.
+ *  Largely inspired by the way Rust does it.
+ *
+ *  The main additions (to be worked out in detail) is a self type `This` and
+ *  a mechanism that a trait can abstract over companions of classes that implement it.
+ *  Companion types and methods are declared using `static` for now, just to
+ *  pick some familiar notation.
+ *
+ *  Ideas about `This` (still vague at the moment)
+ *
+ *   - Treat it as an additional abstract type in a trait, prefixed by the name of the trait
+ *   - An implementing (non-trait) class binds the `This` types of all the traits it implements
+ *     to itself.
+ *   - Later subclasses do not re-bind `This` types already bound by their superclasses.
+ *     (so in that sense, `This`-binding is like trait parameterization, the first implementing
+ *     classes determines the self type and the parameters of a trait)
+ *   - Paramerized classes have parameterized `This` types (e.g. Functor below).
+ */
+import Predef.{any2stringadd => _, _}
+object blueSkyExtensions {
+
+// Semigroup and Monoid
+
+  trait SemiGroup {
+    def + (that: This): This
+  }
+
+  trait Monoid extends SemiGroup {
+    static def unit: This
```

Comment:
Is `static` a separate notion from `common`, or was the keyword changed from `static` to `common` and this should be updated?

Reply:
No, that's just an outdated test (since it is in pending it is not run anyway)